### ISSUE TITLE
[RFC] Migration from deprecated API ->/members to /users

### ISSUE THREAD

**Issue by anishpawaskar** [Real-Dev-Squad/website-backend#1486]
**Why we are making this change?**
- We are planning to deprecate the `/members` API naming convention is not right and it is returning the wrong data it should return members but it’s returning all the users as well as members.

**What is change?**
- We are planning to migrate to `/users` API instead of using `/members.`
- So we will be fetching users and members based on query.
- `/users?member=true` to get the members.
- `/users?member=false` to get the users (not member).

![migrateToUsers](https://github.com/Real-Dev-Squad/website-backend/assets/22213872/43c08a6e-dd47-4d85-8c6d-d400c63e5d10)

**Comment by vinayak-trivedi** [comment]
can we make this generic, like let's suppose we want to allow any role in the query param as true or false, Can we do something like this:

`/users/?role=member&include=true`: returns all the users who have role member as true
`/users/?role=member&include=false`: returns all the users who have role member as false, or don't have that role

this way it can be used for filtering out any role

**Comment by vikhyat187** [comment]
> can we make this generic, like let's suppose we want to allow any role in the query param as true or false, Can we do something like this:
> 
> `/users/?role=member&include=true`: returns all the users who have role member as true `/users/?role=member&include=false`: returns all the users who have role member as false, or don't have that role
> 
> this way it can be used for filtering out any role

Yes completely inline with @vinayak-trivedi's suggestion, say we only want to fetch the `moderators` or `mavens` or `admins` this should be generic enough.

@anishpawaskar  I'm assuming that the `/members` API must be getting used at few other places, so please explore on how the `migration / release plan` should look like and inform the AOs or the people who actively work on those sites and attach the tickets for the migration of those sites using `/members` to `users`

**Comment by anishpawaskar** [comment]
> > can we make this generic, like let's suppose we want to allow any role in the query param as true or false, Can we do something like this:
> > `/users/?role=member&include=true`: returns all the users who have role member as true `/users/?role=member&include=false`: returns all the users who have role member as false, or don't have that role
> > this way it can be used for filtering out any role
> 
> Yes completely inline with @vinayak-trivedi's suggestion, say we only want to fetch the `moderators` or `mavens` or `admins` this should be generic enough.
> 
> @anishpawaskar I'm assuming that the `/members` API must be getting used at few other places, so please explore on how the `migration / release plan` should look like and inform the AOs or the people who actively work on those sites and attach the tickets for the migration of those sites using `/members` to `users`

Sure i will explore it and will inform others as well

**Comment by anishpawaskar** [comment]
> can we make this generic, like let's suppose we want to allow any role in the query param as true or false, Can we do something like this:
> 
> `/users/?role=member&include=true`: returns all the users who have role member as true `/users/?role=member&include=false`: returns all the users who have role member as false, or don't have that role
> 
> this way it can be used for filtering out any role

Sure i will make changes

**Comment by iamitprakash** [comment]
@anishpawaskar this needs more discussion, RFC seems incomplete to me

**Comment by prakashchoudhary07** [comment]
> @anishpawaskar this needs more discussion, RFC seems incomplete to me

+1

**Comment by Ajeyakrishna-k** [comment]
We already have a working API in prod to filter members :` https://api.realdevsquad.com/users/search?role=member`
